rustls/src/conn.rs

```diff
@@ -719,13 +700,177 @@ impl ConnectionCommon {
         state: &mut Option<S>,
         data: &mut S::Data,
     ) -> Result<(), Error> {
+        self.common_state.aligned_handshake = self.handshake_joiner.is_empty();
```

Comment:
While almost all the changes in this commit (in this module) are moving code around, here we use one trick to allow state machine states to inspect the handshake joiner state by caching it as a `bool` in `CommonState`, which then gets checked in `CommonState::check_aligned_handshake()`.